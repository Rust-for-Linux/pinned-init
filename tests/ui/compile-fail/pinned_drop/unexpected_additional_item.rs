use pin_init::*;

#[pin_data(PinnedDrop)]
struct Foo {}

#[pinned_drop]
impl PinnedDrop for Foo {
    fn drop(self: Pin<&mut Self>) {}

    const BAZ: usize = 0;
}

fn main() {}
