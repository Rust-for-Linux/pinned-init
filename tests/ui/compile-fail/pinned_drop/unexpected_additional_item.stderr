error[E0438]: const `BAZ` is not a member of trait `pin_init::PinnedDrop`
  --> tests/ui/compile-fail/pinned_drop/unexpected_additional_item.rs:10:5
   |
10 |     const BAZ: usize = 0;
   |     ^^^^^^^^^^^^^^^^^^^^^ not a member of trait `pin_init::PinnedDrop`

error[E0412]: cannot find type `Pin` in this scope
 --> tests/ui/compile-fail/pinned_drop/unexpected_additional_item.rs:8:19
  |
8 |     fn drop(self: Pin<&mut Self>) {}
  |                   ^^^ not found in this scope
  |
help: consider importing this struct
  |
1 + use std::pin::Pin;
  |
